### ISSUE TITLE
Optimize build times with Input Files list for Xcode 10

### ISSUE THREAD

**Issue by brennanMKE** [realm/SwiftLint#2247]
### Generate a list of included files from SwiftLint configuration

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Feature Request

In the WWDC talk Building Faster in Xcode (linked below) one tip was to include Input Files for a Build Script. It ensures the build only runs the script when those input files have been changed since the previous build. I am finding as I work with a project with several sub-projects which have SwiftLint enabled that the longest part of the build is SwiftLint even when I am not changing those Swift source files.

In the talk a new file type was introduced for `.xcfilelist` which is simply a list of files. This file could be generated by SwiftLint either on the command-line or whenever changes to the list of files is detected. Below is a sample snippet from the talk.

```
# assets.xcfilelist 
# Game

# Asset masters
$(SRCROOT)/Assets/game.psd 
$(SRCROOT)/Assets/player.psd 
$(SRCROOT)/Assets/hud.psd
```

* [Building Faster in Xcode](https://developer.apple.com/videos/play/wwdc2018/408/)

**Comment by brennanMKE** [comment]
Currently I have started using a snippet to disable SwiftLint when I know I just need to run builds and do not want SwiftLint being run each time. I've shared how I do it on an issue related to speeding up the build.

https://github.com/realm/SwiftLint/issues/2129#issuecomment-397905017

**Comment by marcelofabri** [comment]
I'm not sure SwiftLint should handle this. SwiftLint already has caching, which should makes things super fast usually.

A few questions:

* Are you using `--quiet`? Not printing the whole output should speed up.
* Are you using `autocorrect` or just `lint`?
* You've mentioned that there're multiple projects. Are they setup in a way that a file will be checked only one time?

Also, if running SwiftLint takes a lot of time, you can also create an aggregate target to run it (instead of doing what you've mentioned in https://github.com/realm/SwiftLint/issues/2129#issuecomment-397905017) and only run that scheme occasionally.

**Comment by brennanMKE** [comment]
I've never used autocorrect. I will have to try that. I did add `--quiet` now. Using an aggregate target is a good idea too. I could add it in a build scheme too. I will try those changes. Thanks @marcelofabri!

**Comment by marcelofabri** [comment]
Just a note: I wasn't suggesting to use `autocorrect`, was just a question to try to understand the problem 😅

**Comment by brennanMKE** [comment]
Now I just need UXLint to exist and then run autocorrect. ;)

**Comment by marcelofabri** [comment]
I'm closing this since I don't see any actionable items.
